dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-reports-reconciliation-summary.controller/reconciliation-summary-item-mapper.ts

```diff
@@ -0,0 +1,32 @@
+import { Bank, FeeRecordEntity, UtilisationReportEntity } from '@ukef/dtfs2-common';
+import { UtilisationReportReconciliationSummaryItem } from '../../../../types/utilisation-reports';
+
+const getCountOfReconciledFeeRecords = (feeRecords: FeeRecordEntity[]): number => {
+  return feeRecords.filter((feeRecord) => feeRecord.status === 'RECONCILED').length;
+};
+
+const getCountOfDistinctFacilities = (feeRecords: FeeRecordEntity[]): number => {
+  const distinctFacilityIds = new Set(feeRecords.map((feeRecord) => feeRecord.facilityId));
+  return distinctFacilityIds.size;
+};
+
+export const mapReportToSummaryItem = (bank: Bank, report: UtilisationReportEntity): UtilisationReportReconciliationSummaryItem => {
+  const reportReceived = report.status !== 'REPORT_NOT_RECEIVED';
+  const totalFacilitiesReported = getCountOfDistinctFacilities(report.feeRecords);
+  const totalFeesReported = report.feeRecords.length;
+  const reportedFeesLeftToReconcile = totalFeesReported - getCountOfReconciledFeeRecords(report.feeRecords);
```

Comment:
For reports which are marked as complete this will always be the full number of reported fees, which matches the current behaviour, but it seems weird to me that we would display a 'reconciled' report as having fees left to reconcile so I was think we could add a check for if it is reconciled and then just set to 0 (also saves us the calculation for actual reconciled reports i.e. not manual).

@ChristianMcc this has made me think, what are we doing going forwards about reports that were marked as completed? Will be want them to be links to the individual report page in the find reports by year? (and when we first turn the FF on possibly on the bank reports page also) I think currently they will be linked and then actions will show as being possible except they will fail due to the state machine checking report status.

Reply:
Good point I hadn't thought about this before! Going to be quite hard to identify them as we don't have any kind of flag on the report itself, just the state of it's fee records.
In an ideal world I think mark as done behaviour would be updated to set everything as reconciled as well as the utilisation report status (then we would have to run a migration on the old ones that don't have this behaviour), however this feels like way too much overhead for something we won't care about much.

The only place we will be able to get to them is the previous reports/search reports page right?

Reply:
Yes (except for the remainder of the month we turn the feature on if earlier in the month they marked a report as complete) (and then a user could also directly navigate to the individual bank report page if they knew the id and put it in the url directly but I don't think we should worry about this given we expect the api to not accept any calls?)

Reply:
Resolving this thread as PR is ready, will follow up offline on if there are any actions to be taken.